### PR TITLE
[framework] ensure flexible space bar fades when scrolling.

### PR DESCRIPTION
**Pull request by jonahwilliams**
This was missing the actual opacity rendering, oops

### CODE REVIEW

File: packages/flutter/lib/src/material/flexible_space_bar.dart [hunk 1]
```diff
@@ -451,4 +451,22 @@ class _RenderFlexibleSpaceHeaderOpacity extends RenderOpacity {
 
   @override
   bool get isRepaintBoundary => false;
+
+
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/flexible_space_bar.dart:455]: nit: remove one of the duplicated blank lines?

File: packages/flutter/lib/src/material/flexible_space_bar.dart [hunk 2]
```diff
@@ -451,4 +451,22 @@ class _RenderFlexibleSpaceHeaderOpacity extends RenderOpacity {
...
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    if (child == null) {
+      return;
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/flexible_space_bar.dart:459]: Why do we not need to null out the layer in this case (i.e. if you switch from having a child drawn into a layer to not having one)?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/flexible_space_bar.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/material/flexible_space_bar.dart:459]: 🤷 This follows the behavior of the old opacity layer.

File: packages/flutter/lib/src/material/flexible_space_bar.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/material/flexible_space_bar.dart:455]: Done

**Comment by Piinks** [comment]
Did this fix https://github.com/flutter/flutter/issues/128636?

**Comment by sircuri** [comment]
Wondering that too. The following version did not solve that issue:

`Flutter 3.10.6 • channel stable • https://github.com/flutter/flutter.git
Framework • revision f468f3366c (11 hours ago) • 2023-07-12 15:19:05 -0700
Engine • revision cdbeda788a
Tools • Dart 3.0.6 • DevTools 2.23.1`

**Pull request merged**